pkg/gardenlet/bootstrap/bootstrap_test.go

```diff
@@ -31,39 +35,146 @@ import (
+			seedConfig             config.SeedConfig
+
+			approvedCSR = certificatesv1beta1.CertificateSigningRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "watched-csr",
+				},
+				Status: certificatesv1beta1.CertificateSigningRequestStatus{
+					Conditions: []certificatesv1beta1.CertificateSigningRequestCondition{
+						{
+							Type: certificatesv1beta1.CertificateApproved,
+						},
+					},
+					Certificate: []byte("my-cert"),
+				},
+			}
+
+			deniedCSR = certificatesv1beta1.CertificateSigningRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "watched-csr",
+				},
+				Status: certificatesv1beta1.CertificateSigningRequestStatus{
+					Conditions: []certificatesv1beta1.CertificateSigningRequestCondition{
+						{
+							Type: certificatesv1beta1.CertificateDenied,
+						},
+					},
+				},
+			}
 		)
 
 		BeforeEach(func() {
-			ctrl = gomock.NewController(GinkgoT())
-			c = mockclient.NewMockClient(ctrl)
+			secretReference := corev1.SecretReference{
+				Name:      "bootstrap-kubeconfig",
+				Namespace: "garden",
+			}
+
+			// gardenClientConnection with required bootstrap secret kubeconfig secret
+			// in a non-test environment we would use two different secrets
+			gardenClientConnection = &config.GardenClientConnection{
+				BootstrapKubeconfig: &secretReference,
+				KubeconfigSecret:    &secretReference,
+			}
+
+			// seedConfig containing the seed name
+			seed = gardencorev1beta1.Seed{
+				ObjectMeta: metav1.ObjectMeta{Name: seedName},
+			}
+			seedConfig = config.SeedConfig{Seed: seed}
+
+			// create mock seed client
+			s := runtime.NewScheme()
+			Expect(corev1.AddToScheme(s)).ToNot(HaveOccurred())
+			seedClient = fakeclient.NewFakeClientWithScheme(s)
+
+			// rest config for the bootstrap client
+			bootstrapClientConfig = &rest.Config{Host: "testhost", TLSClientConfig: rest.TLSClientConfig{
+				Insecure: false,
+				CAFile:   "filepath",
+			}}
+
+		})
+
+		It("should not return an error", func() {
+			bootstrapClient := fake.NewSimpleClientset(&approvedCSR).CertificatesV1beta1()
+
+			c.EXPECT().Get(ctx, kutil.Key(gardenClientConnection.KubeconfigSecret.Namespace, gardenClientConnection.KubeconfigSecret.Name), gomock.AssignableToTypeOf(&corev1.Secret{}))
+
+			c.EXPECT().Update(ctx, gomock.AssignableToTypeOf(&corev1.Secret{})).DoAndReturn(func(_ context.Context, obj runtime.Object, _ ...client.UpdateOption) error {
+				secret, ok := obj.(*corev1.Secret)
+				Expect(ok).To(BeTrue())
+				Expect(secret.Name).To(Equal(gardenClientConnection.KubeconfigSecret.Name))
+				Expect(secret.Namespace).To(Equal(gardenClientConnection.KubeconfigSecret.Namespace))
+				Expect(secret.Data).ToNot(BeNil())
+				Expect(secret.Data[kubernetes.KubeConfig]).ToNot(BeNil())
+				return nil
+			})
+			c.EXPECT().Delete(ctx, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      gardenClientConnection.KubeconfigSecret.Name,
+					Namespace: gardenClientConnection.KubeconfigSecret.Namespace,
```

Comment:
This test is actually incorrect. Please differentiate between `KubeconfigSecret` and `BootstrapKubeconfig` in `gardenClientConnection` to correct the assertion.
```suggestion
					Name:      gardenClientConnection.BootstrapKubeconfig.Name,
					Namespace: gardenClientConnection.BootstrapKubeconfig.Namespace,
```